SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -82,12 +92,56 @@ - (void)viewDidLoad {
     }
     
+    CGRect searchBarRect = CGRectMake(0, 0, self.view.frame.size.width, kSEARCHBARHEIGHT);
+    self.searchBar = [[UISearchBar alloc] initWithFrame:searchBarRect];
+    self.searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    self.searchBar.placeholder = NSLocalizedString(@"Filter", @"Prompt to search for event names. Here, `Filter` is a joke in English because people filter coffee and this list can be filtered by a term.");
+    self.searchBar.showsCancelButton = true;
+    self.searchBar.delegate = self;
+    
+    CGRect tableSearchViewRect = CGRectMake(0, 0, self.searchBar.frame.size.width, kTABLEHEADERHEIGHT);
+    UIView *tableSearchView = [[UIView alloc] initWithFrame:tableSearchViewRect];
+    tableSearchView.backgroundColor = UIColor.whiteColor;
+    [tableSearchView addSubview:self.searchBar];
+    self.tableView.tableHeaderView = tableSearchView;
+    
+    self.tableView.tableHeaderView.backgroundColor = UIColor.whiteColor;
+    CGPoint contentOffest = self.tableView.contentOffset;
+    contentOffest.y += kSEARCHBARHEIGHT;
+    self.tableView.contentOffset = contentOffest;
+    
+    // TODO: Add an invisible, dismissng button below the search UI
+    
+    self.noResultsView = [[UIView alloc] init];
+    self.noResultsView.frame = CGRectMake(0, (1.5 * kTABLEHEADERHEIGHT), self.tableView.frame.size.width, (self.view.bounds.size.height - (4 * kTABLEHEADERHEIGHT)));
+    [self.noResultsView setHidden:true];
+    [self.view addSubview:self.noResultsView];
+    
+    UILabel *noResultsLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, self.view.frame.size.width, 100)];
+    noResultsLabel.text = NSLocalizedString(@"No events", @"Displayed when no Event names match the given search term");
+    noResultsLabel.textAlignment = NSTextAlignmentCenter;
+    noResultsLabel.font = [UIFont preferredFontForTextStyle:UIFontTextStyleTitle2];
+    [self.noResultsView addSubview:noResultsLabel];
+    
+    [noResultsLabel.topAnchor constraintEqualToAnchor:self.noResultsView.topAnchor].active = true;
+    [noResultsLabel.leftAnchor constraintEqualToAnchor:self.noResultsView.leftAnchor].active = true;
+    [noResultsLabel.rightAnchor constraintEqualToAnchor:self.noResultsView.rightAnchor].active = true;
+    [noResultsLabel.bottomAnchor constraintEqualToAnchor:self.noResultsView.bottomAnchor].active = true;
+    [noResultsLabel.widthAnchor constraintEqualToAnchor:self.noResultsView.widthAnchor].active = true;
+
+    
+    [self.noResultsView.topAnchor constraintEqualToAnchor:self.searchBar.bottomAnchor].active = true;
+    [self.noResultsView.leftAnchor constraintEqualToAnchor:self.tableView.leftAnchor].active = true;
+    [self.noResultsView.rightAnchor constraintEqualToAnchor:self.tableView.rightAnchor].active = true;
+    [self.noResultsView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor].active = true;
+    [self.noResultsView.widthAnchor constraintEqualToAnchor:self.tableView.widthAnchor].active = true;
```

Comment:
I feel like this should be extracted into its own class. I really love the simplicity of the label, but I could see someone having a lot of fun with some wild and crazy animations on this at some point and it would definitely make sense for it to be extracted out for them.